openapi3/media_type.go

```diff
@@ -73,14 +74,37 @@ func (mediaType *MediaType) UnmarshalJSON(data []byte) error {
 
 // Validate returns an error if MediaType does not comply with the OpenAPI spec.
 func (mediaType *MediaType) Validate(ctx context.Context) error {
+	validationOpts := getValidationOptions(ctx)
```

Comment:
Think of the hot path. Move this where it's used `if validationOpts := getValidationOptions(ctx); validationOpts.ExamplesValidationDisabled { ...`